main.nf

```diff
@@ -83,9 +114,92 @@ process CellProfiler {
     file "output/*"
 
   """#!/bin/bash
+mkdir -p output
 
 # Run CellProfiler on this batch of images
 cellprofiler -r -c -o output/ -i input/ -p ${analysis_h5} output/OUTPUT
 
+# Remove the Experiment file
+# Note: this seems fragile, relying on Experiment.txt. 
+#       Check for all non-tabular results instead?
+#       Maybe move the non-tabular results to a separate output folder,
+#       take the first result, and publish the result?
+REMOVE_FILE="\$(ls output/* | grep Experiment.txt)"
+rm \$REMOVE_FILE
+
+# Get the name of the input image file
+export INPUTFILE="\$(ls input/* | head -n 1)"
+export INPUTFILEBASE="\$(basename \$INPUTFILE)"
+
+
+# For all files in the output:
+   * add the image name as the first column
+   * remove the ImageNumber column if it exists
+# Note: the files are renamed to hardcoded temporary files for simplicity
+for f in output/* ; do 
+    # remove carriage returns that are sometimes present
+    cat \$f | tr -d '\\r' > output/tmptrimfile
+    # remove the ImageNumber column if it exists
+    imagenumbercol="\$(head -1 output/tmptrimfile | tr '\\t' '\\n' | cat -n | grep 'ImageNumber' | awk '{print \$1}')"
+    if [[ ! -z "\$imagenumbercol" ]]
+    then
+        cut --complement -f\$imagenumbercol output/tmptrimfile > output/tmpcolfile
+    else
+        cp output/tmptrimfile output/tmpcolfile
+    fi
+    # add the image name column w/ header
+    awk -v f=\$INPUTFILEBASE 'NR==1 {printf("%s\\t%s\\n", "ImageName", \$0)}  NR>1 && NF > 0 { printf("%s\\t%s\\n", f, \$0) }' output/tmpcolfile > output/tmpcopyfile
+    cp output/tmpcopyfile \$f
+    rm -f output/tmpcopyfile
+    rm -f output/tmptrimfile
+    rm -f output/tmpcolfile
+    unset imagenumbercol
+done
+  """
+}
+
+process ConcatFiles_Round1 {
+  container "cellprofiler/cellprofiler:${params.version}"
+  label 'mem_medium'
+  errorStrategy 'retry'
+  maxRetries 3 //slurm tends to SIGTERM jobs
+
+  input:
+    tuple val(filename), path("input???/*")
```

Comment:
Why do you have `???` So interesting!

Reply:
The input file names are all the same, and I suppose I wanted to change the input directory name, rather than the file name. It seems a bit cleaner that way.  The `???` allows for up to 999 input folders, since the directories are named numerically ('input001', 'input002', etc).

Reply:
Do you think there’s a way to do this which doesn’t have a hard limit on 1,000 items?

Reply:
Oh, probably. Maybe path('input*/*'); I'll experiment with that and update the PR once I have something that works.